src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -395,6 +395,26 @@ private void DynamoView_Loaded(object sender, EventArgs e)
             watchSettingsControl.DataContext = background_preview;
         }
 
+        /// <summary>
+        /// Call this method to optionally bring up terms of use dialog. User 
+        /// needs to accept terms of use before any packages can be downloaded 
+        /// from package manager.
+        /// </summary>
+        /// <returns>Returns true if the terms of use for downloading a package 
+        /// is accepted by the user, or false otherwise. If this method returns 
+        /// false, then download of package should be terminated.</returns>
+        /// 
+        bool DisplayTermsOfUseForAcceptance()
+        {
+            var prefSettings = dynamoViewModel.Model.PreferenceSettings;
+            if (prefSettings.PackageDownloadTouAccepted)
+                return true; // User accepts the terms of use.
+
+            var acceptedTermsOfUse = TermsOfUseHelper.ShowTermsOfUseDialog();
+            prefSettings.PackageDownloadTouAccepted = acceptedTermsOfUse;
+            return prefSettings.PackageDownloadTouAccepted; // User may or may not accept the terms.
+        }
+
```

Comment:
@nguyen-binh-minh, I realized this method isn't used anywhere, is the caller left out from your commits?